docs/extensions/readme.md

```diff
@@ -34,6 +40,44 @@ The following documents describes AutoRest specific vendor extensions for [OpenA
 - [x-ms-request-id](#x-ms-request-id) - allows to overwrite the request id header name
 - [x-ms-client-request-id](#x-ms-client-request-id) - allows to overwrite the client request id header name
 
+# Metadata Extensions
+
+## x-ms-secret
+
+This extension is used to annotate the secret property. The value type is boolean and the allowed value is either true or false. Secrets should never expose on a GET.
```

Comment:
@lmazuel ,  can you add below sentence to description? Thanks.
`If a secret does need to be returned after the fact, a POST api can be used to allow for granular RBAC.`